### PR TITLE
Introduce the concept of a large site in order to speed up the Users screen when there are many users

### PR DESCRIPTION
**Pull request by dream-encode**
Trac ticket: https://core.trac.wordpress.org/ticket/38741.

This is almost a 1-1 copy of the most recent patch on Trac.  The relevant version references have been updated to `6.0.0`.

### CODE REVIEW

File: src/wp-includes/functions.php [hunk 1]
```diff
@@ -8409,3 +8409,113 @@ function is_php_version_compatible( $required ) {
...
+ *
+ * @param int|null $network_id ID of the network. Default is the current network.
+ *
+ * @return bool
+ */
+function wp_update_network_user_counts( $network_id = null ) {
+	global $wpdb;
+
+	if ( ! is_multisite() && $network_id ) {
+		 _doing_it_wrong( __FUNCTION__, __( 'Unable to pass $nework_id if not using multisite.' ), '5.0.0' );
```
Comment (spacedmonkey) [inline src/wp-includes/functions.php]:
```suggestion
		 _doing_it_wrong( __FUNCTION__, __( 'Unable to pass $nework_id if not using multisite.' ), '6.0.0' );
```

Comment (dream-encode) [reply src/wp-includes/functions.php]: @spacedmonkey Thanks!  There's a small typo there, too.

**Comment by dream-encode** [comment]
Looks like tests are failing.  The first big issues I see is "Unexpected deprecated notice for wp_is_large_network".  We'll need to rework the tests to account for the deprecation.

**Comment by dream-encode** [comment]
Merged in https://core.trac.wordpress.org/changeset/53011.

**Pull request closed without merging**